backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -438,5 +438,37 @@ public async Task<IActionResult> DeleteVideos([FromBody] List<Guid> ids)
 
             return Ok();
         }
+
+        [HttpGet, Route("Recycled")]
+        public async Task<IActionResult> AllRecycledVideos()
+        {
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
```

Comment:
`            var user = await _usersRepository.GetUserById(userId);
            if (user is null)
            {
                return NotFound();
            }`

Reply:
some boilerplate code is missing